content/browser/web_contents/web_contents_view_child_frame.cc

```diff
@@ -48,25 +49,48 @@ WebContentsViewChildFrame::WebContentsViewChildFrame(
 
 WebContentsViewChildFrame::~WebContentsViewChildFrame() = default;
 
-WebContentsView* WebContentsViewChildFrame::GetOuterView() {
+WebContentsImpl* WebContentsViewChildFrame::GetHostingWebContents() {
+  if (auto* outer_web_contents = web_contents_->GetOuterWebContents()) {
+    return outer_web_contents;
+  }
+  if (auto* secure_embed_delegate = web_contents_->GetSecureEmbedDelegate()) {
+    return static_cast<WebContentsImpl*>(
+        secure_embed_delegate->GetEmbedderWebContents());
+  }
+  return nullptr;
```

Comment:
optional: `return const_cast<WebContentsViewChildFrame*>(this)->GetHostingWebContents();`